### PR TITLE
Check plan is not lock before auto_execution

### PR DESCRIPTION
**Pull request by iNecas**
There is a potential race condition, where a plan is still in planned state,
but some executor has already locked it.

### CODE REVIEW

File: lib/dynflow/coordinator.rb [hunk 1]
```diff
@@ -146,6 +146,14 @@ def initialize(world)
         @data.merge!(owner_id: "world:#{world.id}",  world_id: world.id)
       end
 
+      def self.lock_id(*args)
+        riase NoMethodError
```
Comment (adamruzicka) [inline lib/dynflow/coordinator.rb]: typo

**Comment by adamruzicka** [comment]
Any tips on how to trigger this race condition?

**Comment by iNecas** [comment]
The easiest it to follow https://bugzilla.redhat.com/show_bug.cgi?id=1344707#c11

File: lib/dynflow/coordinator.rb (hunk 1 above)
Comment (iNecas) [inline lib/dynflow/coordinator.rb]: Updated

**Comment by adamruzicka** [comment]
Seems to really fix the issue, ACK

**Pull request merged**